src/transport.hpp

```diff
@@ -47,6 +47,7 @@ class Transport {
 	virtual bool send(message_ptr message) = 0;
 
 	void onRecv(message_callback callback) { mRecvCallback = std::move(callback); }
+	bool recvCallbackValid() { return mRecvCallback; }
 
 protected:
 	void recv(message_ptr message) { mRecvCallback(message); }
```

Comment:
You should do the check directly here actually rather than introducing a new method.

However, this test is already done there: https://github.com/paullouisageneau/libdatachannel/blob/8530b20dbea34d27e52ae1e8a4aeda0d51c11725/include/rtc/include.hpp#L77

Reply:
Actually it was not making its job but somehow now it seems ok. I am investigating.